moz_security/io_modules/taskcluster.lua

```diff
@@ -0,0 +1,68 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+--[[
+# Heka Message Extensions for Taskcluster
+
+This module is intended to be used with another IO module such as a decoder to
+add a username field if the client is a human user.
+
+## Functions
+
+### add_username
+
+Given a Heka message with a Taskcluster clientid, attempt to parse out the username
+associated with the clientid. This is currently specific to Mozilla LDAP usernames,
+as that is the primary way in which human users authenticate with Taskcluster. If no
+username is found, do nothing.
+
+*Arguments*
+- msg (table) - original message
+- field_name (string) - field name in the message to lookup
+
+*Return*
+- none - the message is modified in place if a username is found
+
+## Configuration examples
+taskcluster_decoder = {
+    username_field = "username",
+    matchers = {
+      "'mozilla-auth0/ad|Mozilla-LDAP|' (s <- {%w+}) '/'"
+    }
+}
+decoder_module = {
+    {
+        { "decoders.heka.table_to_fields" },
```

Comment:
This proposed syntax is ambiguous and conflicts with the grammar builder syntax.

Reply:
Agreed. Going to wait till https://github.com/mozilla-services/lua_sandbox_extensions/issues/386 is fleshed out as this is taken from the syntax that @ameihm0912 was using before creating #381

Reply:
381 is in (the first argument is a sting since decoders don't take any additional options)
```lua
decoder_module = { {"decoders.heka.table_to_fields", {clientid = "taskcluster#add_username"}} }
```